client/lib/query-manager/theme/index.js

```diff
@@ -1,16 +1,131 @@
 /**
+ * External dependencies
+ */
+import { cloneDeep, get, isEqual, keyBy, range } from 'lodash';
+
+/**
  * Internal dependencies
  */
 import PaginatedQueryManager from '../paginated';
 import ThemeQueryKey from './key';
-import { isThemeMatchingQuery } from './util';
 import { DEFAULT_THEME_QUERY } from './constants';
 
 /**
  * ThemeQueryManager manages themes which can be queried
  */
 export default class ThemeQueryManager extends PaginatedQueryManager {
-	matches = isThemeMatchingQuery;
+	/**
+	 * Signal that an item(s) has been received for tracking. Optionally
+	 * specify that items received are intended for patch application, or that
+	 * they are associated with a query. This function does not mutate the
+	 * instance state. Instead, it returns a new instance of ThemeQueryManager if
+	 * the tracked items have been modified, or the current instance otherwise.
+	 *
+	 * Note that we implement our own receive() method instead of just relying on
+	 * that of the base class and implementing matches() and sort() to make use of
+	 * it. This is because our themes query REST API endpoint uses ElasticSearch
+	 * heavily to sort results by relevancy. Mimicking ES's complex weighting
+	 * algorithms on the client side would be non-trivial, or would require a
+	 * heavy (and possibly slow) dependency such as elasticlunr.
+	 * By implementing receive() ourselves instead, we can override the base class
+	 * QueryManager's sorting behavior and keep the order as returned from the endpoint.
```

Comment:
This comment is a bit verbose I think. We'd be crazy to implement some ES equivalent in the client, since we have a really fast ES endpoint that returns themes in the exact order we want them.

All we need to say is that the endpoint order is correct already. I think QueryManager exists because some data (posts?) needs to jump around? We don't have that problem, so we don't need to explain why we're not solving it.